dora/src/bytecode/generator.rs

```diff
@@ -488,6 +488,77 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
         }
     }
 
+    fn visit_expr_delegation(&mut self, expr: &ExprDelegationType, dest: DataDest) -> Register {
+        let call_type = self.src.map_calls.get(expr.id).unwrap().clone();
+        let fct_id = call_type.fct_id().unwrap();
+
+        let fct = self.vm.fcts.idx(fct_id);
+        let fct = fct.read();
+
+        let callee_id = if fct.kind.is_definition() {
+            unimplemented!()
+        } else {
+            fct_id
+        };
+
+        let callee = self.vm.fcts.idx(callee_id);
+        let callee = callee.read();
+
+        if let FctKind::Builtin(_intrinsic) = callee.kind {
+            unimplemented!()
+        }
+
+        let return_type = if dest.is_effect() {
+            BuiltinType::Unit
+        } else {
+            self.specialize_type_for_call(&call_type, callee.return_type)
+        };
+        let arg_types = callee
+            .params_with_self()
+            .iter()
+            .map(|&arg| self.specialize_type_for_call(&call_type, arg).into())
+            .collect::<Vec<BytecodeType>>();
+        let num_args = arg_types.len();
+
+        let return_reg = if return_type.is_unit() {
+            Register::invalid()
+        } else {
+            self.ensure_register(dest, return_type.into())
+        };
+
+        let start_reg = if num_args > 0 {
```

Comment:
There should always be at least one register, right? Since self is always passed as first argument.

Reply:
(At least here in delegation)

Reply:
Should i add an assertion to guarantee our assumption?